File: Binary Search/1. BasicBinarySearch/BasicBinarySearch.py
class Solution(object):
    def binary_search(self, array, target):
        if array == None or len(array) == 0:
            return -1
        lo, hi = 0, len(array) - 1
        while lo <= hi:
            mid = lo + (hi - lo) // 2
            n = array[mid]
            if target == n:
                return mid
            elif target < n:
                hi = mid - 1
            else:
                lo = mid + 1
        return -1

File: Binary Search/1. BasicBinarySearch/BasicBinarySearch.java
class Solution {
    public int binarySearch(int[] array, int target) {
        if (array == null || array.length == 0) {
            return -1;
        }
        int lo = 0, hi = array.length - 1;
        while (lo <= hi) {
            int mid = lo + (hi - lo) / 2;
            int n = array[mid];
            if (target == n) {
                return mid;
            } else if (target < n) {
                hi = mid - 1;
            } else {
                lo = mid + 1;
            }
        }
        return -1;
    }
}